src/ui/components/RecipeEnrollmentPage.tsx

```diff
@@ -1,38 +1,129 @@
       setJsonInput(event.target.value);
     },
-    [],
+    [setJsonInput],
   );
 
-  const handleEnrollClick = useCallback(async () => {
-    try {
-      const result = await browser.experiments.nimbus.enrollInExperiment(
-        JSON.parse(jsonInput) as object,
-      );
+  const handleEnrollClick = useCallback(
+    async (
+      event?: React.MouseEvent<HTMLButtonElement>,
+      forceEnroll = false,
+    ) => {
+      if (!jsonInput) {
+        addToast({ message: "Invalid Input: Enter JSON", variant: "danger" });
+        return;
+      }
 
-      if (result) {
-        addToast({ message: "Enrollment successful", variant: "success" });
-        setJsonInput("");
-      } else {
-        addToast({ message: "Enrollment failed", variant: "danger" });
+      let parsedRecipe;
+      try {
+        parsedRecipe = JSON.parse(jsonInput) as object;
+      } catch (error) {
+        addToast({
+          message: `Error enrolling into experiment: ${
+            (error as Error).message ?? String(error)
+          }`,
+          variant: "danger",
+        });
+        return;
       }
-    } catch (error) {
-      addToast({
-        message: `Error enrolling into experiment: ${(error as Error).message ?? String(error)}`,
-        variant: "danger",
-      });
+
+      try {
+        const result = await browser.experiments.nimbus.enrollInExperiment(
+          parsedRecipe as object,
+          forceEnroll,
+        );
+
+        if (result.enrolled) {
+          addToast({ message: "Enrollment successful", variant: "success" });
+          setJsonInput("");
+        } else if (result.error) {
+          setEnrollError(result.error);
+        }
+      } catch (error) {
+        addToast({
+          message: `Error enrolling into experiment: ${
+            (error as Error).message ?? String(error)
+          }`,
+          variant: "danger",
+        });
+      }
+    },
+    [jsonInput, setJsonInput, addToast, setEnrollError],
+  );
+
+  const handleModalConfirm = useCallback(async () => {
+    setEnrollError(null);
+    await handleEnrollClick(undefined, true);
+  }, [handleEnrollClick, setEnrollError]);
+
+  const handleModalClose = useCallback(() => {
+    setEnrollError(null);
+  }, [setEnrollError]);
+
+  function EnrollmentError(
+    slug: string,
+    enrollError: EnrollInExperimentResult["error"],
+  ) {
+    if (!enrollError) {
+      return null;
     }
-  }, [jsonInput, setJsonInput, addToast]);
+    const { activeEnrollment, slugExistsInStore } = enrollError;
+
+    if (activeEnrollment && slugExistsInStore) {
+      return (
+        <p>
+          There is already an enrollment for the slug: <strong>{slug}</strong>.
+          Would you like to proceed with force enrollment by unenrolling,
+          deleting, and re-enrolling into the new configuration?
+        </p>
+      );
+    }
+
+    if (activeEnrollment) {
+      return (
+        <p>
+          There is an active enrollment for the slug: <strong>{slug}</strong>.
```

Comment:
This actually might not be the same slug.

```suggestion
          There is an active enrollment for the slug: <strong>{activeEnrollment}</strong>.
```